lib/api/MerchantsApi.php

```diff
@@ -0,0 +1,367 @@
+        HeaderSelector $selector = null
+    ) {
+        $this->client = $client ?: new Client();
+        $this->config = $config ?: new Configuration();
+        $this->headerSelector = $selector ?: new HeaderSelector();
+    }
+
+    /**
+     * @return Configuration
+     */
+    public function getConfig()
+    {
+        return $this->config;
+    }
+
+    /**
+     * Operation readMerchant
+     *
+     * Read merchant
+     *
+     * @param  string $merchant_guid The unique identifier for a &#x60;merchant&#x60;. (required)
+     *
+     * @throws \atrium\ApiException on non-2xx response
+     * @throws \InvalidArgumentException
+     * @return \atrium\model\MerchantResponseBody
+     */
+    public function readMerchant($merchant_guid)
+    {
+        list($response) = $this->readMerchantWithHttpInfo($merchant_guid);
+        return $response;
+    }
+
+    /**
+     * Operation readMerchantWithHttpInfo
+     *
+     * Read merchant
+     *
+     * @param  string $merchant_guid The unique identifier for a &#x60;merchant&#x60;. (required)
+     *
+     * @throws \atrium\ApiException on non-2xx response
+     * @throws \InvalidArgumentException
+     * @return array of \atrium\model\MerchantResponseBody, HTTP status code, HTTP response headers (array of strings)
+     */
+    public function readMerchantWithHttpInfo($merchant_guid)
+    {
+        $returnType = '\atrium\model\MerchantResponseBody';
+        $request = $this->readMerchantRequest($merchant_guid);
+
+        try {
+            $options = $this->createHttpClientOption();
+            try {
+                $response = $this->client->send($request, $options);
+            } catch (RequestException $e) {
+                throw new ApiException(
+                    "[{$e->getCode()}] {$e->getMessage()}",
+                    $e->getCode(),
+                    $e->getResponse() ? $e->getResponse()->getHeaders() : null,
+                    $e->getResponse() ? $e->getResponse()->getBody()->getContents() : null
+                );
+            }
+
+            $statusCode = $response->getStatusCode();
+
+            if ($statusCode < 200 || $statusCode > 299) {
+                throw new ApiException(
+                    sprintf(
+                        '[%d] Error connecting to the API (%s)',
+                        $statusCode,
+                        $request->getUri()
+                    ),
+                    $statusCode,
+                    $response->getHeaders(),
+                    $response->getBody()
+                );
+            }
+
+            $responseBody = $response->getBody();
+            if ($returnType === '\SplFileObject') {
+                $content = $responseBody; //stream goes to serializer
+            } else {
+                $content = $responseBody->getContents();
+                if ($returnType !== 'string') {
+                    $content = json_decode($content);
+                }
+            }
+
+            return [
```

Comment:
Returning an array like this is a bit weird. What is the rationale behind it?

Reply:
I'm not sure. That is what the default swagger generator produced. It follows that same pattern on every endpoint in the library.